[docs] Add Workflows and Maestro insights docs (#46447)
expo/expo: docs/pages/eas-insights/workflows.mdx

---

> title: Workflows insights

Reviewer: ```suggestion
title: EAS Workflows insights
```

Correct service name use.

> sidebar_title: Workflows

Reviewer: ```suggestion
sidebar_title: EAS Workflows
```

description: Track run counts, success rates, and trends for your EAS Workflows in the Workflows tab of EAS Insights.
---

import { Dataflow03Icon } from '@expo/styleguide-icons/outline/Dataflow03Icon';

import { BoxLink } from '~/ui/components/BoxLink';
import { ContentSpotlight } from '~/ui/components/ContentSpotlight';

> **important** **EAS Insights** is in [preview](/more/release-statuses/#preview) and subject to breaking changes.

> **info** Workflows insights is available on the Production and Enterprise plans. Learn more at [EAS pricing](https://expo.dev/pricing).

> The **Workflows** tab shows how your [EAS Workflows](/eas/workflows/introduction/) are doing over time: how often they run, how often they succeed, and which workflows fail most. Data appears automatically as your workflows run — there's nothing to install. Open your project in the EAS dashboard, select **Insights** from the navigation menu, and then select the **Workflows** tab.

Reviewer: **[suggestion]** The style guide prefers separate sentences over an em dash when splitting a thought (`expo-docs-style-guide.md#punctuation`).

```suggestion
The **Workflows** tab in the dashboard shows how your [EAS Workflows](/eas/workflows/introduction/) are doing over time: how often they run, how often they succeed, and which workflows fail most. Data appears automatically as your workflows run. 

- To view these insights, open your project in the EAS dashboard, select **Insights** from the navigation menu, and then select the **Workflows** tab.
```

> Choose a time range to load the data. Every metric also compares against the previous period of equal length so you can see whether a number is trending up or down.

Reviewer: ```suggestion
- Choose a time range to load the data. Every metric also compares against the previous period of equal length so you can see whether a number is trending up or down.
```

I'd like us to tightly couple these as bullet points here since they relate to one instruction: to reach the end goal of how can i view workflow insights for my Expo project.

<ContentSpotlight
  alt="The Workflows tab in EAS Insights showing run metrics, a runs-over-time chart, and a table of workflows."
  src="/static/images/eas-insights/insights-workflows-light.png"
  darkSrc="/static/images/eas-insights/insights-workflows-dark.png"
/>

## Overview

The top of the tab summarizes activity for the selected time range:

- **Total runs**: how many workflow runs started.
- **Success rate**: the share of runs that succeeded.
- **Active workflows**: how many distinct workflows ran.

> - **Failed runs**: how many runs failed.

Reviewer: ```suggestion
- **Total runs**: How many workflow runs started.
- **Success rate**: The share of runs that succeeded.
- **Active workflows**: How many distinct workflows ran.
- **Failed runs**: How many runs failed.
```

## Runs over time

A chart breaks runs down by day (or by hour or minute for shorter time ranges) into total, successful, failed, and canceled runs. Toggle between a line view and a stacked view, and turn individual statuses on or off to focus on what you care about.

## Workflows table

Below the chart, a table lists each workflow that ran in the time range with:

- **Runs**: the total count, broken down into succeeded, failed, and canceled.
- **Success rate**: the share of runs that succeeded.

> - **Last run**: when the workflow last ran.

Reviewer: ```suggestion
- **Runs**: The total count, broken down into succeeded, failed, and canceled.
- **Success rate**: The share of runs that succeeded.
- **Last run**: When the workflow last ran.
```

## Filters

Narrow the data down by workflow, run status (success, failure, canceled), trigger type (manual, schedule, or a GitHub event), or git ref. You can also search by workflow name.

## Export

> Export the workflow runs that match your current filters and time range as CSV or NDJSON for further analysis.

Reviewer: **[nit]** Spell out NDJSON on first mention; it is not in the known-abbreviations list (`expo-docs-style-guide.md#word-usage`).

```suggestion
Export the workflow runs that match your current filters and time range as CSV or newline-delimited JSON (NDJSON) for further analysis.
```
